app/components/key_pair_generator_component.js

```diff
@@ -0,0 +1,39 @@
+import { trackEvent } from '@18f/identity-analytics';
+
+class KeyPairGeneratorElement extends HTMLElement {
+  async generateKeyPair() {
+    const t0 = performance.now();
+    const keypair = await crypto.subtle.generateKey(
+      {
+        name: 'RSA-OAEP',
+        modulusLength: 4096,
+        publicExponent: new Uint8Array([0x01, 0x00, 0x01]),
+        hash: 'SHA-256',
+      },
+      true,
+      ['encrypt', 'decrypt'],
+    );
+
+    const encodedPrivate = await crypto.subtle.exportKey('pkcs8', keypair.privateKey);
+    this.privateB64key = btoa(String.fromCharCode.apply(null, new Uint8Array(encodedPrivate)));
+
+    const encodedPublic = await crypto.subtle.exportKey('spki', keypair.publicKey);
+    this.publicB64key = btoa(String.fromCharCode.apply(null, new Uint8Array(encodedPublic)));
+    const t1 = performance.now();
```

Comment:
[suggestion] these repeated computations might justify their own utility function. Then it will be obvious they are doing the same thing with different inputs as well as what that input is. Right now you kind of have to hunt for it.

Reply:
Agreed! Done in 5c9830193

Reply:
So the typescript linter didn't like how I broke it out, so for now I reverted it as it was.